File: 01_Programming_Basics/03_Programming_Basics_with_JavaScript/05_While_Loop/07_WaterDispenser.js
function WaterDispenser(args) {
  let water = parseFloat(args[0]);
  let i = 0;
  let buttonPushed = "";

  while(water > 0) {
    i++;
    buttonPushed = args[i];

    if (buttonPushed == "Easy") { water -= 50; }
    else if (buttonPushed == "Medium") { water -= 100; }
    else if (buttonPushed == "Hard") { water -= 200; }
  }

    if (water < 0) {
      console.log(Math.abs(water) + "ml has been spilled.")
    } else {
      console.log("The dispenser has been tapped " + i + " times.");
    }
    
}

File: 01_Programming_Basics/01_Programming_Basics_with_C++/08_While_Loop_Lab/07_Water_Dispenser.cpp
#include <iostream>
#include <iomanip>
#include <string>
using namespace std;

int main() {
 
  int water = 0;
  int counter = 0;
  string input = "";
  cin >> water;

  while (true) {
    cin >> input;

    if (input == "Hard") {
      water -= 200;
    } else if (input == "Medium") {
      water -= 100;
    } else if (input == "Easy") {
      water -= 50;
    }
    counter++;

    if (water <= 0) { break; }

  }

  if (water < 0 ) {
    cout << abs(water) << "ml has been spilled." << endl;
  } else {
    cout << "The dispenser has been tapped " << counter << " times." << endl;
  }

}